CONTRIBUTING.md

```diff
@@ -1,47 +1,199 @@
-3. Ensure local tests pass.
-4. Commit to your fork using clear commit messages.
-5. Send us a pull request, answering any default questions in the pull request interface.
-6. Pay attention to any automated CI failures reported in the pull request, and stay involved in the conversation.
+As more and more modules are introduced to AWS Amplify, it became a necessity to modularize the library into smaller pieces so that users could avoid importing unnecessary parts into their app. The goal of this design is to make AWS Amplify modularized and also keep it backward compatible to avoid breaking changes.
 
-GitHub provides additional document on [forking a repository](https://help.github.com/articles/fork-a-repo/) and
-[creating a pull request](https://help.github.com/articles/creating-a-pull-request/).
+Modular import prevents unnecessary code dependencies are included with the app, and thus decreases the bundle size and enables adding new functionality without the risk of introducing errors related to the unused code.
 
+Amplify has established the concepts of categories and plugins. A category is a collection of api calls that are exposed to the client to do things inside that category. For example, in the storage category generally one wants to upload and download objects from storage so the apis exposed to the client will represent that functionality. Because Amplify is pluggable, a plugin of your choosing will provide the actual implementation behind that api interface. Using the same example of Storage, the plugin we choose might be AWSStoragePlugin which would then implement each api call from the category with a service call or set of service calls to S3, the underlying storage provider of the AWS plugin.
+
+## Getting Started
+
+To get started with contributing to the iOS library of Amplify, first make sure you have the latest version of Xcode installed as well as cocoapods. You can install cocoapods by simply running:
+
+```
+sudo gem install cocoapods
+```
+
+Then make sure you fork the project first and then clone it by running:
+
+```
+git clone git@github.com:YOURGITHUBUSERNAME/amplify-ios.git
+```
+GitHub provides additional documentation on [forking a repository](https://help.github.com/articles/fork-a-repo/).
+
+The project itself is broken up into various Xcode projects so it's important to understand what you want to change and open the correct project. At the root of the project there is an Xcode project/workspace called Amplify. The Amplify project contains all the models and interfaces that we expose for each category so all the generic parts of a category live here. If you wanted for instance to add another API to the Storage category, you might start by running `pod install` right at the root and opening the Amplify workspace and heading to the Storage folder like below:
```

Comment:
It would be better to be more specific than saying it's broken-up into various projects.  Also might wanna use the word "partitioned", "separated", or "divided" instead of "broken" -- it makes it sound like things are broken :P.

That being said, maybe something like this would be more direct:

The Amplify iOS framework has been divided into multiple project files:

Core - `Amplify.xcworkspace` - Includes high level category interfaces and shared interfaces.  Shared components including Hub.
API - `AmplifyPlugins/API/APICategoryPlugin.xcworkspace`
Analytics - `AmplifyPlugins/Analytics/AnalyticsCategoryPlugin.xcworkspace`
DataStore - `AmplifyPlugins/DataStore/DataStoreCategoryPlugin.xcworkspace`
Predictions - `AmplifyPlugins/Predictions/PredictionsCategoryPlugin.xcworkspace`
Storage - `AmplifyPlugins/Storage/StoragePlugin.xcworkspace`

Prior to making changes, you will need to run `pod install` in the directory that you intend to make changes in.  For example, to modify Core:

Reply:
changed